spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -139,6 +142,25 @@ public void eventListenersAreAutoRegistered() {
 				.onPreSave(any(Event.class));
 	}
 
+	@Test
+	public void autoIndexConfiguration() {
+		load(CustomConfigurationFactory.class);
+		assertThat(this.context.getBean(org.neo4j.ogm.config.Configuration.class)
+				.getAutoIndex()).isEqualTo(AutoIndexMode.NONE);
+
+		load(CustomConfigurationFactory.class, "spring.data.neo4j.auto-index=assert");
```

Comment:
The point I am trying to make is that if you set a property to check that it is created properly, you don't need to test all the possible values. If you do you are testing the binder, not your code.